### PR TITLE
feat: add directive to project dropdowns through cdk portal

### PR DESCRIPTION
**Pull request by steveblue**
## Summary

This addresses a long standing issue where dropdowns are stuck in the layer in the stack where they are defined, opening up the possibility for scrollable containers or overflow: none to crop parts of the dropdown. Added an optional Directive for projecting ngx-dropdown through CDK Portal. It requires adding the Directive and importing some global CSS for ease of use. Added documentation and used the Directive in JSON Editor.

Screenshot 2026-07-30 at 1 39 39 PM

## Checklist

- [ ] \*Added unit tests
- [ ] \*Added a code reviewer
- [ ] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [ ] Updated the demo page
- [ ] Included screenshots of visual changes

_\*required_

---

> [!NOTE]
> **Medium Risk**
> Touches shared dropdown overlay lifecycle (scroll listeners, detach/close sync) and changes default JSON Editor menu DOM placement; regressions would show as mispositioned menus or flaky E2E, not auth/data risk.
> 
> **Overview**
> Adds an **opt-in** `ngxDropdownPortal` directive on `ngx-dropdown` that moves the menu into an Angular CDK overlay (`DomPortal`) so menus are not clipped by `overflow` on ancestors. Configurable offsets, `portalPanelClass`, and `portalFollowScroll` (menu tracks the trigger on scroll; closes when the trigger leaves the viewport or a scroll parent). Global styles ship via `dropdown-portal.scss` in the library CSS bundle; the global-styles build adds `node_modules` to Sass include paths for CDK overlay mixins.
> 
> **JSON Editor** type-picker and options menus (flat and classic tree) now use `ngxDropdownPortal` with `ngx-json-editor-dropdown-panel` so add-property / add-item / node-options menus render above scrollable editor UI.
> 
> **Tests & docs:** Cypress JSON editor specs query menus in `.cdk-overlay-container`; the dropdown demo page adds a scroll-clipping portal example and API table for the new inputs. `DropdownPortalDirective` is exported from `public_api`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b9160ef877141a8028766ea16432e1fdc2ba65d8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      this.overlayRef.updateScrollStrategy(this.createScrollStrategy());
+    }
+
+    this.overlayRef.updatePositionStrategy(this.createPositionStrategy(menu));
+    this.overlayRef.attach(new DomPortal(menu));
+
+    if (this.portalFollowScroll) {
+      this.bindScrollListeners();
+    }
+  }
```
[the remark this answers is not in the document]
Comment (steveblue) [reply projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts:79]: mouseleave is not closing the menu in practice.

File: projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-dropdown-panel.scss [hunk 2]
```diff
@@ -0,0 +1,5 @@
+@use './json-editor.extensions';
+
+.ngx-json-editor-dropdown-panel .ngx-dropdown-menu {
+  @extend %ngx-dropdown-menu;
+}
```
[the remark this answers is not in the document]
Comment (steveblue) [reply projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-dropdown-panel.scss:5]: the margin looks fine.

File: projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+  private getPanelClasses(): string[] {
+    const classes = ['ngx-dropdown', 'open', 'adjusted', 'ngx-dropdown-portal-panel'];
+
+    if (!this.portalPanelClass) {
+      return classes;
+    }
+
+    return classes.concat(Array.isArray(this.portalPanelClass) ? this.portalPanelClass : [this.portalPanelClass]);
+  }
```
[the remark this answers is not in the document]
Comment (steveblue) [reply projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts:246]: The caret is no longer part of the design.

File: projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      },
+      {
+        originX,
+        originY: 'top',
+        overlayX,
+        overlayY: 'bottom',
+        offsetX,
+        offsetY
+      }
+    ];
```
[the remark this answers is not in the document]
Comment (steveblue) [reply projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts:235]: Seems to shift top or bottom fine.

File: projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    const isOpen = this.dropdown.open;
+
+    if (isOpen && !this.wasOpen) {
+      this.attach();
+    } else if (!isOpen && this.wasOpen) {
+      this.detach();
+    }
+
+    this.wasOpen = isOpen;
+  }
```
[the remark this answers is not in the document]
Comment (steveblue) [reply projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-portal.directive.ts:49]: Dont see this in practice.

**Review by jogaj — APPROVED** [review]
